core/inspector.rs

```diff
@@ -315,89 +309,53 @@ impl JsRuntimeInspector {
             // function was called while session were polled asynchronously.
             w.poll_state = prev_state;
           });
-
           break;
         }
-        PollState::BlockOnWaitingForSession => {
+        PollState::Blocking => {
           futures::executor::block_on(futures::future::poll_fn(|cx| {
-            // Accept new connections.
-            let poll_result = sessions.session_rx.poll_next_unpin(cx);
-            if let Poll::Ready(Some(session_proxy)) = poll_result {
-              let mut session = InspectorSession::new(
-                self.v8_inspector.clone(),
-                session_proxy,
-                false,
-              );
-
-              if let Poll::Ready(Some(session_stream_item)) =
-                session.poll_next_unpin(cx)
-              {
-                let (v8_session_ptr, msg) = session_stream_item;
-                InspectorSession::dispatch_message(v8_session_ptr, msg);
+            loop {
+              // Accept new connections.
+              let poll_result = sessions.session_rx.poll_next_unpin(cx);
+              match poll_result {
+                Poll::Ready(Some(session_proxy)) => {
+                  let session = InspectorSession::new(
+                    self.v8_inspector.clone(),
+                    session_proxy,
+                    false,
+                  );
+                  sessions.established.push(session);
+                  continue; // Poll `session_rx` until it returns `Pending`.
+                }
+                Poll::Ready(None) => unreachable!(), // Is it really unreachable?.
```

Comment:
Yes, we hold both ends of the channel on inspector, so this arm shouldn't ever hit